mp/src/vgui2/vgui_controls/CvarComboBox.cpp

```diff
@@ -43,6 +43,29 @@ int CvarComboBox::AddItem(const char* itemText, const KeyValues* userData)
     return iRtnVal;
 }
 
+void CvarComboBox::OnKeyCodeTyped(KeyCode code)
+{
+    BaseClass::OnKeyCodeTyped(code);
+    // updates cvar to text in combobox when up/down keys are used
+    int activeItem = GetActiveItem();
+    switch (code)
+    {
+    case KEY_DOWN:
+        activeItem++;
+        activeItem %= GetItemCount();
+        break;
+    case KEY_UP:
+        activeItem--;
+        if (activeItem < 0)
+            activeItem = GetItemCount() - 1;
+        break;
+    default: break;
+    }
+    ActivateItemByRow(activeItem);
+    PostActionSignal(new KeyValues("ControlModified"));
```

Comment:
Does ActivateItemByRow not trigger OnTextChanged?

Reply:
It does, didn't need these `PostActionSignal` / `ApplyChanges()` calls.

Reply:
Originally I tried having this functionality just in `OnTextChanged()`, but couldn't find a function to give me what the actual value inside the textentry is. I was using `GetMenu()->GetCurrentlyHighlightedItem()` which worked apart from the arrow keys not always moving on to the correct next item (eg. up on MPH giving UPS, etc.).